### PR TITLE
Ability to use updater script to pin dev requirements

### PR DESCRIPTION
**Pull request by AlanCoding**
##### SUMMARY
Don't plan to use this here, but it's good to have for forks, so they won't get surprised by updates to these libraries.

##### ISSUE TYPE
 - Bug, Docs Fix or other nominal change

##### COMPONENT NAME
 - API

### CODE REVIEW

File: requirements/requirements_dev.txt [hunk 1]
```diff
@@ -9,7 +9,6 @@ black
 pytest!=7.0.0
 pytest-cov
 pytest-django
-pytest-pythonpath
```
Comment (AlanCoding) [inline requirements/requirements_dev.txt:13]: This library requires an old version of pytest which is, of course, conflicting with other libraries here. I guess it just ignored the conflict up until this point. If you actually go to the project, it basically says it's old and don't use it anymore. This won't work with the conflict, and it seems we didn't need this either.

File: requirements/updater.sh [hunk 2]
```diff
@@ -67,6 +79,7 @@ main() {
     echo "help      Print this message"
     echo "run       Run the process only upgrading pinned libraries from requirements.in"
     echo "upgrade   Upgrade all libraries to latest while respecting pinnings"
+    echo "dev       Pin the development requirements file"
```
Comment (jbradberry) [inline requirements/updater.sh:82]: The `echo "Usage: $0 [run|upgrade]"` line above ought to also be updated.

**Review by jbradberry — APPROVED** [review]
This seems good to me, aside from my pretty minor picky comment.

**Comment by chrismeyersfsu** [comment]
@TheRealHaoLiu had the idea of pinning the black version when we do a tower release. I like that. Keep awx devel unpinned so we hit new versions of black and deal with the changes. Pin old releases to black version at that time.

This change unlocks that ability.

**Review by chrismeyersfsu — APPROVED** [review]

**Comment by AlanCoding** [comment]
I re-tested and it works so I'll go ahead and merge.

**Pull request merged**